### PR TITLE
Cog 1010 disable submit buttons

### PR DESCRIPTION
**Pull request by Sushmareddy-OSI**
**Related story** 
[COG-1010](https://osi-cwds.atlassian.net/browse/COG-1010)

**Brief feature description**
Submit buttons are enabled in "Login Page", "Password Reset Page" and "Update Password Page"  only when all required input text fields have values

### CODE REVIEW

File: src/components/LoginForm.test.js [hunk 1]
```diff
@@ -186,10 +186,19 @@ describe('LoginForm.js Tests', () => {
       expect(button).toHaveLength(1)
       expect(button.props().onClick).toEqual(mockToVerify)
     })
-    it('check if submit button is disabled or not', () => {
-      const button = wrapper.find('button')
-      expect(button.props().disabled).toEqual(disableSignIn)
+
+    describe('check if Submit button is disabled or not', () => {
+      it('checks when Submit button is not disabled', () => {
+        const wrapper = shallow(<LoginForm email="a@test.com" password="password" disableSignIn={false}/>)
```
Comment (Nasir2712) [inline src/components/LoginForm.test.js]: since you are using email and password check on disable button, please remove the disableSignIn prop and just pass email and password to see if it's disabled or not.

File: src/components/ResetPasswordForm.test.js [hunk 2]
```diff
@@ -180,6 +180,18 @@ describe('ResetPasswordForm.js Tests', () => {
       expect(button.at(0).text()).toEqual('Change Password')
     })
 
+    describe('check if Change Password button is disabled or not', () => {
+      it('checks when Change Password button is not disabled', () => {
+        const wrapper = shallow(<ResetPasswordForm email="a@test.com" code="the_code" newPassword="new_password" confirmPassword="confirm_password"/>)
+        expect(wrapper.find('button').at(0).props().disabled).toEqual(false)
+      })
+
+      it('checks when Change Password button is disabled', () => {
```
Comment (Nasir2712) [inline src/components/ResetPasswordForm.test.js]: I think a separate describe blocks with different props need to be tested here.

File: src/components/LoginForm.test.js (hunk 1 above)
Comment (Sushmareddy-OSI) [reply src/components/LoginForm.test.js]: I didn't removed disableSignIn because the text is changing to "Loading.." onClick. I am not sure if it is a business requirement or not, if not I can remove.

File: src/components/ResetPasswordForm.test.js (hunk 2 above)
Comment (Sushmareddy-OSI) [reply src/components/ResetPasswordForm.test.js]: sure! will update

File: src/components/LoginForm.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ const LoginForm = ({onSubmit, errorMsg, successMessage, email, password,
       <label className='label-customizable' htmlFor='password'>Password</label>
       <PasswordInput id='password' placeholder='Password' password={password} onChange={onPasswordChange}/>
       <a className='redirect-customizable' href='/forgotPassword'>Forgot your password?</a>
-      <button className="btn btn-primary submitButton-customizable" id="submit" type="submit" onClick={onSubmit} disabled={disableSignIn}>
+      <button className="btn btn-primary submitButton-customizable" id="submit" type="submit" onClick={onSubmit} disabled={!(email && password)}>
```
Comment (Nasir2712) [inline src/components/LoginForm.js]: Also, the logic here depends on the props being an empty string, can we add default props to all the files where they are being used, just to make sure that undefined case has been taken care of.

File: src/components/LoginForm.test.js (hunk 1 above)
Comment (Nasir2712) [reply src/components/LoginForm.test.js]: only for this test case, is the test case failing if you don't pass that prop?

File: src/components/ResetPasswordForm.test.js (hunk 2 above)
Comment (Nasir2712) [reply src/components/ResetPasswordForm.test.js]: can you also add a test scenario where one of the props is null?
